src/utils/logger.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2021 John Mifsud <imabuddha@gmail.com>
+ *
+ * This file is part of Touchégg.
+ *
+ * Touchégg is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License  as  published by  the  Free Software
+ * Foundation,  either version 3 of the License,  or (at your option)  any later
+ * version.
+ *
+ * Touchégg is distributed in the hope that it will be useful,  but  WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the  GNU General Public License  for more details.
+ *
+ * You should have received a copy of the  GNU General Public License along with
+ * Touchégg. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef UTILS_LOGGER_H_
+#define UTILS_LOGGER_H_
+
+#include <string>
+
+/**
+ * Singleton class to provide simple logging functionality.
+ */
+class Logger {
+ public:
+  enum class LogLevel {
+    INFO,
+    WARNING,
+    ERROR,
+    DEBUG,
+    GESTURE_INFO,
```

Comment:
Why GESTURE_INFO and UPDATE_INFO? I think that we could use `debug` for gesture update info and `info` for gesture info, as the app name and the gesture detection is useful information

Reply:
Since my free time window is limited & you hadn't replied to my questions in the issue I had to make some assumptions. My guiding principle is to **keep the program behavior completely backward compatible**.

Since the dbg lvl msgs are only displayed when `--verbose` is specified, using debug for gesture update msgs would change behavior. **I'm fine with this if you are.** In fact it simplifies things as I could remove the `--no-update-messages` flag as those wouldn't appear by default anymore.

However, my motivation for doing this effort is to be able to suppress all non-error/warning gesture msgs, as these are only useful during setup/troubleshooting. Constantly emitting these msgs during normal use just clogs up my `.xsession-errors` log file making any real issues harder to see.

Since **I need this functionality** then one of the following is necessary, either:
* all normal (not error or warning) gesture msgs use debug lvl
* we still require the separate `gesture` log lvl (because otherwise the << func has no way to know a msg is a gesture msg)

If you choose the first option then the `no-gesture-messages` flag can also be removed. This would be great IMHO. However it also changes current behavior. (user would have to run touchegg client with `--verbose` or '-v' flags **and** '--client` to see gesture msgs)

The second option is also OK with me.

Reply:
Hi! Sorry for not asking your questions in the issue, but you created the pull request in 1 day since your first question and I didn't have time to answer there, so I moved the conversation here.

I'm fine removing by default every message and showing only warning and error messages unless `--verbose` is set. Not sure about `-v`, as it is commonly used for version.

I'll keep the `info` level for:

```
Gesture begin detected
Gesture information:
[...]
```

And `--debug` for issue reporting information, like the update %, just in case we want to add extra flags in the future.